numpy/_core/fromnumeric.py

```diff
@@ -1507,6 +1508,14 @@ def searchsorted(a, v, side='left', sorter=None):
     array([0, 5, 1, 2])
 
     """
+    dtype_a = np.asarray(a).dtype
+    dtype_v = np.asarray(v).dtype
```

Comment:
```suggestion
    a = np.asanyarray(a)
    v = np.asanyarray(v)
    dtype_a = a.dtype
    dtype_v = v.dtype
```
This should reduce the performance impact of this PR. I will look int the C implementation as well, as there might be some improvements possible there as well.